### PR TITLE
[Keyboard] Add hineybush/h87_g2

### PR DESCRIPTION
**Pull request by hineybush**
## Description

Adding firmware for my new "Gen 2" h87, which is STM32-based

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/hineybush/h87_g2/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#include QMK_KEYBOARD_H
```
Comment (waffle87) [inline keyboards/hineybush/h87_g2/keymaps/default/keymap.c:1]: This file is missing a license header.
eg.
```
// Copyright 2023 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/hineybush/h87_g2/info.json [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    },
+    "processor": "STM32F072",
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0001",
+        "vid": "0x4048"
+    },
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (waffle87) [inline keyboards/hineybush/h87_g2/info.json:27]: This change will also need to be reflected in the keymaps.
```suggestion
        "LAYOUT": {
```

File: keyboards/hineybush/h87_g2/readme.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# h87_g2
+
+![h87_g2](imgur.com image replace me!)
+
+New generation of the h87 keyboard PCB platform with an STM32 microcontroller.
+
+* Keyboard Maintainer: [Josh Hinnebusch](https://github.com/hineybush)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
```
Comment (waffle87) [inline keyboards/hineybush/h87_g2/readme.md:9]: These fields should be filled out.

**Review by waffle87 — CHANGES_REQUESTED** [review]
There are also submodule commits that need to be reverted. This [doc](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) may help.

File: keyboards/hineybush/h87_g2/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+// Copyright 2023 Josh Hinnebusch
+// SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (zvecr) [inline keyboards/hineybush/h87_g2/keymaps/default/keymap.c:3]:
```suggestion
// Copyright 2023 Josh Hinnebusch
// SPDX-License-Identifier: GPL-2.0-or-later
#include QMK_KEYBOARD_H
```

File: keyboards/hineybush/h87_g2/readme.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# h87_g2
+
+[h87_g2](https://i.imgur.com/t7chDf8.png)
```
Comment (drashna) [inline keyboards/hineybush/h87_g2/readme.md:3]:
```suggestion
[h87_g2](https://i.imgur.com/t7chDf8h.png)
```

**Review by drashna — APPROVED** [review]

File: keyboards/hineybush/h87_g2/readme.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# h87_g2
+
+[h87_g2](https://i.imgur.com/t7chDf8h.png)
+
+New generation of the h87 keyboard PCB platform with an STM32 microcontroller.
+
+* Keyboard Maintainer: [Josh Hinnebusch](https://github.com/hineybush)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
```
Comment (waffle87) [inline keyboards/hineybush/h87_g2/readme.md:9]:
```suggestion
* Hardware Supported: H87 G2 PCB w/ STM32F072 MCU
* Hardware Availability: [hineybush.com](https://hineybush.com)
```

**Pull request merged**